packages/vow/src/watch.js

```diff
@@ -81,6 +81,7 @@ const preparePromiseWatcher = (zone, isRetryableReason, watchNextStep) =>
       const state = {
         vow: /** @type {unknown} */ (undefined),
         priorRetryValue: /** @type {any} */ (undefined),
+        seenPayloads: zone.detached().weakSetStore('seenPayloads'),
```

Comment:
Is this the right way? Is there some scaling considerations about creating a new short lived weak store for every `watch` operation?

Reply:
Oh, wait, the zone is durable, so this is making a new durable WeakSetStore. Yeah, that's got some significant overhead, we allocate a new collection ID (eg a vref) for each one. When the Store is GCed, we'll delete all that, but we still have several DB reads+writes for each one. It would be a lot better to make a single WeakSetStore for the whole vat, and have all watch operations share it.

Reply:
the set is specific to each watch, so I don't see how to avoid making a new set per watch...

Reply:
The alternative is to keep the vows strongly and use a copy array.